tests/src/kakarot/accounts/test_contract_account.py

```diff
@@ -123,3 +124,41 @@ def test_should_read_bytecode(self, cairo_run, bytecode, storage):
             calls = [call(address=address) for address in addresses]
             mock_storage.assert_has_calls(calls)
             assert output[:output_len] == list(bytecode)
+
+    class TestUpgrade:
+        @SyscallHandler.patch(
+            "IAccount.library_call_version", lambda class_hash, data: [2000]
+        )
+        @patch.object(SyscallHandler, "mock_replace_class", mock.MagicMock())
```

Comment:
may not be the best way to do it but there might also be something with the `SycallHandler.patch` decorator wrecking the usual flow

Reply:
I'm not sure I get what you mean

Reply:
https://docs.python.org/3/library/unittest.mock.html#patch I'd have expected something like

```
@patch('tests.utils.syscall_handler.SyscallHandler.mock_replace_class')
def test_name(self, mock_replace_class):
```

but it doesn't seem to work well, so there may be something with the `@SyscallHandler` decorator breaking the `@patch` one.

In any case, I think that the `mock.MagicMock()` param is not required, see https://docs.python.org/3/library/unittest.mock.html#patch-object